homeassistant/components/gandi_livedns/__init__.py

```diff
@@ -0,0 +1,228 @@
+"""Integrate with Gandi Live DNS service."""
+import asyncio
+from datetime import timedelta
+import logging
+
+import aiohttp
+from aiohttp.hdrs import AUTHORIZATION
+import async_timeout
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_API_KEY,
+    CONF_DOMAIN,
+    CONF_NAME,
+    CONF_TIMEOUT,
+    CONF_TTL,
+    CONF_TYPE,
+)
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = "gandi_livedns"
+
+INTERVAL = timedelta(minutes=5)
+
+CONF_IPV6 = "ipv6"
+
+DEFAULT_TIMEOUT = 10
+DEFAULT_TTL = 3600
+DEFAULT_TYPE = "A"
+DEFAULT_IPV6 = False
+
+AVAILABLE_TYPE = [
+    "A",
+    "AAAA",
+    "ALIAS",
+    "CAA",
+    "CDS",
+    "CNAME",
+    "DNAME",
+    "DS",
+    "KEY",
+    "LOC",
+    "MX",
+    "NAPTR",
+    "NS",
+    "OPENPGPKEY",
+    "PTR",
+    "RP",
+    "SPF",
+    "SRV",
+    "SSHFP",
+    "TLSA",
+    "TXT",
+    "WKS",
+]
+
+IPV4_PROVIDER_URL = "https://api.ipify.org"
+IPV6_PROVIDER_URL = "https://api6.ipify.org"
+
+GANDI_LIVEDNS_API_URL = (
+    "https://api.gandi.net/v5/livedns/domains/{domain}/records/{rrname}/{rrtype}"
+)
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required(CONF_DOMAIN): cv.string,
+                vol.Required(CONF_API_KEY): cv.string,
+                vol.Required(CONF_NAME): cv.string,
+                vol.Optional(CONF_TYPE, default=DEFAULT_TYPE): vol.In(AVAILABLE_TYPE),
+                vol.Optional(CONF_TTL, default=DEFAULT_TTL): cv.positive_int,
+                vol.Optional(CONF_IPV6, default=DEFAULT_IPV6): cv.boolean,
+                vol.Optional(CONF_TIMEOUT, default=DEFAULT_TIMEOUT): cv.positive_int,
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
```

Comment:
For integrations that connect to devices or services, we no longer accept new YAML configurations.

This integration needs to be adjusted to use a configuration flow and config entries.

More information about this can be found in Architecture Decision Record:

- ADR-0010: <https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md#decision>

See our developer documentation on how to get started creating a configuration flow for this integration:

<https://developers.home-assistant.io/docs/config_entries_config_flow_handler>

Thanks already! :+1:

Reply:
Hi,
I had not seen this resolution 
I have made changes to add flow for  config entries. i hope I did everything right  :)